File: 64-minimum-path-sum/64-minimum-path-sum.js
var minPathSum = function(grid) {
    let height = grid.length;
    let width = grid[0].length;
    
    let table = new Array(height+1).fill().map(()=> new Array(width+1).fill(Infinity));
    
    table[height-1][width-1] = grid[height-1][width-1];
    // console.log(table);
    
    for(let row=height-1;row>=0;row--){
        for( let col=width-1;col>=0;col--){
            if(row===height-1 && col===width-1) continue;
            table[row][col] = grid[row][col] + Math.min(table[row+1][col],table[row][col+1]);
        }
    }

    // console.log(table);
    return table[0][0];
};

File: 64-minimum-path-sum/64-minimum-path-sum.py
class Solution:
    def minPathSum(self, grid: List[List[int]]) -> int:
        for subarray in grid:
            subarray.append(float('inf'))
        grid.append([float('inf')]*len(grid[0]))
        
        
        for row in range(len(grid)-2,-1,-1):
            for col in range(len(grid[0])-2,-1,-1):
                if row==len(grid)-2 and col==len(grid[0])-2: continue
                grid[row][col]+=min(grid[row][col+1],grid[row+1][col])
                
                
        # print(grid)
        return grid[0][0]